examples_tests/41.VisibilityBuffer/main.cpp

```diff
@@ -176,40 +157,90 @@ void main()
 #define _IRR_FRAG_MAIN_DEFINED_
 )";
 
-
 // TODO: return original fragment and vertex shader GLSL (for future embedding in compute)
-core::smart_refctd_ptr<asset::ICPUSpecializedShader> createModifiedFragShader(const asset::ICPUSpecializedShader* _fs)
+core::smart_refctd_ptr<asset::ICPUSpecializedShader> createModifiedShader(const asset::ICPUSpecializedShader* _shader, const char* _shaderSpecificOverride)
 {
-    const asset::ICPUShader* unspec = _fs->getUnspecialized();
+    const asset::ICPUShader* unspec = _shader->getUnspecialized();
     assert(unspec->containsGLSL());
 
     std::string glsl = reinterpret_cast<const char*>(unspec->getSPVorGLSL()->getPointer());
-    size_t firstNewlineAfterVersion = glsl.find("\n", glsl.find("#version "));
-    glsl.insert(firstNewlineAfterVersion, GLSL_FRAG_OVERRIDE);
+    glsl.insert(glsl.find("\n", glsl.find("#version")), GLSL_COMMON_OVERRIDE);
+    glsl.insert(glsl.find("\n", glsl.find("#version")), _shaderSpecificOverride);
```

Comment:
won't that make you insert the shader specific before the common?

Reply:
i dont think so, why?

Reply:
you've inserted the `GLSL_COMMON_OVERRIDE` right after the first newline after `#version`

then you do the same thing with `_shaderSpecificOverride`

which would insert the last before the first.